threads/src/main/kotlin/pl/edu/pw/ia/threads/domain/query/projector/ThreadProjector.kt

```diff
@@ -52,6 +58,28 @@ class ThreadProjector(
 		repository.delete(event.threadId)
 	}
 
+	@EventHandler
+	fun on(event: AccountUpdatedEvent) {
+		val views = repository.findByAccountId(event.accountId)
+			.map { it.copy(username = event.name) }
+			.subscribe {repository.save(it)}
+	}
+
+	@EventHandler
+	fun on(event: PostCreatedEvent) {
+		val view = repository.findById(event.threadId) ?: throw ThreadNotFoundException(event.threadId)
+		repository.save(view.copy(postId = event.postId, post = event.content))
+	}
+
+	@EventHandler
+	fun on(event: PostUpdatedEvent) {
+		val postView = queryGateway.query(FindPostByIdQuery(event.postId), PostView::class.java).get()
+		val threadView = repository.findById(postView.threadId) ?: throw ThreadNotFoundException(postView.threadId)
```

Comment:
findByIdAndPostId(event.threadId, event.postId)

Reply:
and do not throw any exception in case of thread not found
most of updated posts won't be first in a thread and this exception will be thrown most of the time